lib/einhorn/command.rb

```diff
@@ -123,10 +123,14 @@ def self.signal_all(signal, children=nil, record=true)
         end
       end
 
-      if Einhorn::State.signal_timeout
+      if Einhorn::State.signal_timeout && record
         Einhorn::Event::Timer.open(Einhorn::State.signal_timeout) do
           children.each do |child|
             next unless spec = Einhorn::State.children[child]
+            unless spec[:signaled].include?(signal)
```

Comment:
Yeah, I just thought of this too. I'll look into making `spec[:signaled]` hold more context.